src/core/ui.js

```diff
@@ -34,6 +34,21 @@ function ariaDecorate(elem, ariaMap) {
   }, elem);
 }
 
+const observer = new IntersectionObserver(
+  entries => {
+    entries.forEach(entry => {
+      if (entry.intersectionRatio > 0) {
```

Comment:
Can we use [`IntersectionObserverEntry.isIntersecting`](https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry/isIntersecting)?

Reply:
No, as it reports if it’s intercepting inside the viewport.